pkg/controller/clone-controller.go

```diff
@@ -310,6 +311,14 @@ func (r *CloneReconciler) updatePvcFromPod(sourcePod *corev1.Pod, pvc *corev1.Pe
 	return nil
 }
 
+func sourcePodFinished(sourcePod *corev1.Pod) bool {
+	if sourcePod == nil {
+		return true
+	}
+
+	return sourcePod.Status.Phase == corev1.PodSucceeded || sourcePod.Status.Phase == corev1.PodFailed
```

Comment:
Why should also corev1.PodFailed will result pvc.Annotations[AnnCloneOf] = "true"?

Reply:
It won't. The first element of the condition, `if podSucceededFromPVC(pvc)...` will allow setting the annotation only if the cloning succeeds.

Reply:
So why not drop the PodFailed condition? to support the generic "finished" although there is currently only one caller of this func?

Reply:
Because it would change the logic. If the cloning pod would fail, the cloning process will never end.